2002-08-26  Jakub Jelinek  <jakub@redhat.com>

	* lib/verdef.h (xlt_verdef): Don't crash if calculating size of
	section which hasn't been loaded yet.
	* lib/verneed.h (xlt_verneed): Likewise.

--- libelf-0.8.2/lib/verdef.h.jj	2001-10-07 22:03:02.000000000 +0200
+++ libelf-0.8.2/lib/verdef.h	2002-08-26 19:04:54.000000000 +0200
@@ -138,6 +138,9 @@ xlt_verdef(unsigned char *dst, const uns
     if (n < sizeof(verdef_stype)) {
 	return 0;
     }
+    if (dst == NULL && src == NULL) {
+	return n;
+    }
     soff = doff = 0;
     for (;;) {
 	const verdef_stype *svd;
--- libelf-0.8.2/lib/verneed.h.jj	2001-10-07 22:03:02.000000000 +0200
+++ libelf-0.8.2/lib/verneed.h	2002-08-26 19:05:18.000000000 +0200
@@ -142,6 +142,9 @@ xlt_verneed(unsigned char *dst, const un
     if (n < sizeof(verneed_stype)) {
 	return 0;
     }
+    if (dst == NULL && src == NULL) {
+	return n;
+    }
     soff = doff = 0;
     for (;;) {
 	const verneed_stype *svn;
